### PR TITLE
[audio] Add initial docs for alpha release

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Adds the initial version of the `expo-audio` docs for the alpha release. 

# How
Same approach as what we did for expo-video. The Expo AV audio doc now appear in the sidebar as `Audio (expo-av)` and `Audio` directs to the doc for the new library.

# Test Plan

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/audio.mdx [hunk 1]
```diff
@@ -2,35 +2,71 @@
...
 ---
 
 import APISection from '~/components/plugins/APISection';
 import { APIInstallSection } from '~/components/plugins/InstallSection';
+import { ConfigPluginExample, ConfigPluginProperties } from '~/ui/components/ConfigSection';
 import { SnackInline } from '~/ui/components/Snippet';
 import { PlatformTags } from '~/ui/components/Tag';
 
-`Audio` from `expo-av` allows you to implement audio playback and recording in your app.
+> **Info** This page documents an upcoming version of the Audio library. **Expo audio is currently in alpha and subject to breking changes.**
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/audio.mdx:16]:
```suggestion
> **warning** This page documents an upcoming version of the Audio library. **Expo audio is currently in alpha and subject to breaking changes.**
```

Warning is an appropriate callout here.

File: docs/pages/versions/unversioned/sdk/audio-av.mdx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+- MediaRecorder encoding options and other configurations are inconsistent across browsers, utilizing a Polyfill such as [kbumsik/opus-media-recorder](https://github.com/kbumsik/opus-media-recorder) or [ai/audio-recorder-polyfill](https://github.com/ai/audio-recorder-polyfill) in your application will improve your experience. Any options passed to `prepareToRecordAsync` will be passed directly to the MediaRecorder API and as such the polyfill.
+- Web browsers require sites to be served securely for them to listen to a mic. See [MediaDevices `getUserMedia()` security](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia#security) for more details.
+
+## API
+
+```js
+import { Audio } from 'expo-av';
+```
+
+<APISection packageName="expo-audio" apiName="Audio" />
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/audio-av.mdx:193]:
```suggestion
<APISection packageName="expo-audio-av" apiName="Audio" />
```

To fetch the correct reference as per the change in `GenerateDocsAPIData.ts`, lets update the package name here.

File: docs/pages/versions/unversioned/sdk/audio-av.mdx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+iconUrl: '/static/images/packages/expo-av.png'
+platforms: ['android', 'ios', 'tvos', 'web']
+---
+
+import APISection from '~/components/plugins/APISection';
+import { APIInstallSection } from '~/components/plugins/InstallSection';
+import { SnackInline } from '~/ui/components/Snippet';
+import { PlatformTags } from '~/ui/components/Tag';
+
+`Audio` from `expo-av` allows you to implement audio playback and recording in your app.
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/audio-av.mdx:15]:
```suggestion
> **info** The `Audio` component from `expo-av` documented on this page will be replaced by an improved version in `expo-audio` in an upcoming release when the new library is stable. [Learn about `expo-audio`](https://docs.expo.dev/versions/unversioned/sdk/audio/).

`Audio` from `expo-av` allows you to implement audio playback and recording in your app.
```

Maybe we should add a callout here about the upcoming Audio library, just like we do for Video [(reference)](http://localhost:3002/versions/unversioned/sdk/audio/).

CleanShot 2024-10-23 at 22 40 29

**Review by amandeepmittal — CHANGES_REQUESTED** [review]

File: docs/pages/versions/unversioned/sdk/audio.mdx [hunk 4]
```diff
@@ -2,35 +2,71 @@
 title: Audio
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/audio.mdx:2]: One thing that I found annoying in Video is the lack of the package name here. Of course we'll remove it once `expo-av` docs are removed.
```suggestion
title: Audio (expo-audio)
```

**Comment by clarkg** [comment]
@alanjhughes et. al any chance you guys could pls provide an example of recording a sound and then playing back the same recording with the new API?

the documentation has examples for recording and playing separately, not together

When I tried it:
- when the recording ends, recording.uri is reset
- and there's no method to load new URI into AudioPlayer after your instantiate it the first time

**Comment by alanjhughes** [comment]
@clarkg our example screen does exactly that https://github.com/expo/expo/blob/main/apps/native-component-list/src/screens/Audio/expo-audio/RecordingScreen.tsx

**Comment by clarkg** [comment]
ah doh ty @alanjhughes

**Comment by alanjhughes** [comment]
@clarkg no worries! The library is in early beta so I didn't want to fill the docs with examples that could go out of date quickly, things will likely change as we get feedback from users but to will add more examples for the stable release in sdk 53. Any feedback you have is welcome. Feel free to ping me on discord etc.

**Pull request merged**